skaha/src/main/java/org/opencadc/skaha/session/UserStorageClient.java

```diff
@@ -0,0 +1,366 @@
+                        // Create the folder if not already present.
+                        final ContainerNode sslHiddenFolderNode = new ContainerNode(".ssl");
+                        final URI sslFolderURI = URI.create(userStorageConfiguration.userHomeBaseURI
+                                + sslHomeFolder.toAbsolutePath().toString());
+                        final Node createdNode = cavernClient.createNode(new VOSURI(sslFolderURI), sslHiddenFolderNode);
+                        if (!(createdNode instanceof ContainerNode)) {
+                            throw new IllegalStateException("BADNESS: Created Node is not a ContainerNode: "
+                                    + createdNode
+                                    + ".  Expected ContainerNode called " + sslHiddenFolderNode.getName() + " at "
+                                    + userStorageConfiguration.userHomeBaseURI.getPath());
+                        } else {
+                            LOGGER.debug("Created SSL folder: " + sslHomeFolder.toAbsolutePath());
+                        }
+                    }
+
+                    final String certificateFileName = "cadcproxy.pem";
+                    final DataNode certificateNode = new DataNode(certificateFileName);
+                    final CredClient credClient = new CredClient(credServiceID);
+                    final Subject currentSubject = AuthenticationUtil.getCurrentSubject();
+                    final X509CertificateChain proxyCert = Subject.callAs(
+                            CredUtil.createOpsSubject(),
+                            () -> credClient.getProxyCertificate(currentSubject, UserStorageClient.ONE_WEEK_DAYS));
+
+                    LOGGER.debug("Proxy cert: " + proxyCert);
+                    final Path userCertPath =
+                            Path.of(owner, UserStorageClient.PROXY_CERTIFICATE_FOLDER_NAME, certificateFileName);
+                    final URI certificateNodeURI =
+                            URI.create(userStorageConfiguration.userHomeBaseURI + "/" + userCertPath);
+                    final VOSURI targetVOSURI = new VOSURI(certificateNodeURI);
+                    final Node injectedCertificateNode = cavernClient.createNode(targetVOSURI, certificateNode);
+                    final File certificateFile =
+                            new File(Path.of(UserStorageClient.PROXY_CERTIFICATE_FOLDER_NAME, certificateFileName)
+                                    .toString());
+
+                    try (final FileOutputStream fileOutputStream = new FileOutputStream(certificateFile);
+                            final InputStream certificateInputStream = new ByteArrayInputStream(
+                                    proxyCert.certificateString().getBytes())) {
+                        IOUtils.copy(certificateInputStream, fileOutputStream);
+                        fileOutputStream.flush();
+                    }
```

Comment:
Who / what is creating the .ssl directory? 

On a new user that directory doesn’t exist so the catch-all logs a warning, and certificate deployment fails?